### PR TITLE
planner: fix bug in `PhysicalTableReader::Clone()`

### LINKED ISSUE #45299: parallel apply does not work with mpp
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
mysql> drop table if exists test.t;
mysql> create table test.t (a int, b int);
mysql> alter table test.t set tiflash replica 1;
mysql> insert into test.t values(1,2),(2,1),(10,15),(15,10),(20,25),(25,10),(18,19),(19,18);
mysql> insert into test.t select a + 5, b + 5 from test.t;
mysql> insert into test.t select a + 5, b + 5 from test.t;
mysql> insert into test.t select a + 5, b + 5 from test.t;
mysql> insert into test.t select a + 5, b + 5 from test.t;
mysql> insert into test.t select a + 5, b + 5 from test.t;
mysql> insert into test.t select a + 5, b + 5 from test.t;
mysql> insert into test.t select a + 5, b + 5 from test.t;
mysql> insert into test.t select a + 5, b + 5 from test.t;
mysql> insert into test.t select a + 5, b + 5 from test.t;
```
```
mysql> explain select count(*) from t where a > (select a from t t1 where t.a > t1.a limit 1);
+--------------------------------------+---------+--------------+---------------+----------------------------------------------------------+
| id                                   | estRows | task         | access object | operator info                                            |
+--------------------------------------+---------+--------------+---------------+----------------------------------------------------------+
| HashAgg_15                           | 1.00    | root         |               | funcs:count(1)->Column#7                                 |
| └─Apply_18                           | 4091.90 | root         |               | CARTESIAN inner join, other cond:gt(test.t.a, test.t.a)  |
|   ├─TableReader_25(Build)            | 4091.90 | root         |               | MppVersion: 2, data:ExchangeSender_24                    |
|   │ └─ExchangeSender_24              | 4091.90 | mpp[tiflash] |               | ExchangeType: PassThrough                                |
|   │   └─Selection_23                 | 4091.90 | mpp[tiflash] |               | not(isnull(test.t.a))                                    |
|   │     └─TableFullScan_22           | 4096.00 | mpp[tiflash] | table:t       | pushed down filter:empty, keep order:false, stats:pseudo |
|   └─Selection_26(Probe)              | 3273.52 | root         |               | not(isnull(test.t.a))                                    |
|     └─Limit_29                       | 4091.90 | root         |               | offset:0, count:1                                        |
|       └─TableReader_39               | 4091.90 | root         |               | MppVersion: 2, data:ExchangeSender_38                    |
|         └─ExchangeSender_38          | 4091.90 | mpp[tiflash] |               | ExchangeType: PassThrough                                |
|           └─Limit_37                 | 4091.90 | mpp[tiflash] |               | offset:0, count:1                                        |
|             └─Selection_36           | 4091.90 | mpp[tiflash] |               | gt(test.t.a, test.t.a)                                   |
|               └─TableFullScan_35     | 5114.88 | mpp[tiflash] | table:t1      | pushed down filter:empty, keep order:false, stats:pseudo |
+--------------------------------------+---------+--------------+---------------+----------------------------------------------------------+
mysql> set tidb_enable_parallel_apply=0;
mysql> select count(*) from t where a > (select a from t t1 where t.a > t1.a limit 1);
+----------+
| count(*) |
+----------+
|     4095 |
+----------+
1 row in set (0.78 sec)
mysql> set tidb_enable_parallel_apply=1;
mysql> select count(*) from t where a > (select a from t t1 where t.a > t1.a limit 1);
+----------+
| count(*) |
+----------+
|        0 |
+----------+
1 row in set (0.78 sec)
```
### 2. What did you expect to see? (Required)
When enable parallel apply, the query return correct result
### 3. What did you see instead (Required)
When enable parallel apply, query result is wrong
### 4. What is your TiDB version? (Required)
master@ 90dabc0433a75db650ac683897d264ef58d63a73

### PR DESCRIPTION
**Pull request by windtalker**
### What problem does this PR solve?

Issue Number: close #45299

Problem Summary:

### What is changed and how it works?
Fix `PhysicalTableReader::Clone`, so `PhysicalTableReader::TablePlans` are still extracted from `PhysicalTableReader::tablePlan`
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: planner/core/physical_plans.go [hunk 1]
```diff
@@ -269,9 +269,8 @@ func (p *PhysicalTableReader) Clone() (PhysicalPlan, error) {
 	if cloned.tablePlan, err = p.tablePlan.Clone(); err != nil {
 		return nil, err
 	}
-	if cloned.TablePlans, err = clonePhysicalPlan(p.TablePlans); err != nil {
-		return nil, err
-	}
+	// TablePlans are actually the flattened plans in tablePlan, so can't copy them, just need to extract from tablePlan
+	cloned.TablePlans = flattenPushDownPlan(cloned.tablePlan)
```
[the remark this answers is not in the document]
Comment (windtalker) [reply planner/core/physical_plans.go:273]: Yes, I think you are right, and it does not cause a bug because `indexPlan` is not used during execution time.

**Review by SeaRise — APPROVED** [review]

File: planner/core/physical_plans.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (SeaRise) [reply planner/core/physical_plans.go:273]: ok

**Pull request merged**